types/time.go

```diff
@@ -978,7 +978,7 @@ func (d Duration) ToNumber() *MyDecimal {
 		signStr = "-"
 	}
 
-	if d.Fsp == 0 {
```

Comment:
I don't think we should modify this function. We can use checkfsp() to modify fsp when it is UnspecifiedFsp.

Reply:
Comment addressed, PTAL, Thanks. @wshwsh12